src/wallet/receive.cpp

```diff
@@ -313,10 +313,23 @@ bool CWalletTx::IsTrusted() const
     return pwallet->IsTrusted(*this, trusted_parents);
 }
 
-CWallet::Balance CWallet::GetBalance(const int min_depth, const bool avoid_reuse, const bool fAddLocked, const CCoinControl* coinControl) const
+CAmount CWallet::GetBalanceAnonymized(const CCoinControl& coinControl) const
+{
+    if (!CCoinJoinClientOptions::IsEnabled()) return 0;
+
+    CAmount anonymized_amount{0};
+    LOCK(cs_wallet);
+    for (auto pcoin : GetSpendableTXs()) {
+        anonymized_amount += pcoin->GetAnonymizedCredit(coinControl);
+    }
+    return anonymized_amount;
```

Comment:
could probably use accumulate
```suggestion
LOCK(cs_wallet);
return std::accumulate(GetSpendableTXs().begin(), GetSpendableTXs().end(), CAmount{0},
    [&](CAmount sum, const auto& pcoin) {
        return sum + pcoin->GetAnonymizedCredit(coinControl);
    });```

Reply:
should not use std::accumulate() without ranges, I think.

GetSpendableTXs() does caclucation inside, it doesn't just return reference to internal structure.
So, begin() and end() can even belong to 2 different data structures in memory.